### PR TITLE
Add efs_file_system_backup_policy resource

### LINKED ISSUE #14372: Add backup support for aws_efs_file_system
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for automatic backups for EFS. See [announcement](https://aws.amazon.com/about-aws/whats-new/2020/07/announcing-automatic-backups-for-amazon-elastic-file-system/).

### New or Affected Resource(s)

* aws_efs_file_system

### Potential Terraform Configuration

```hcl
resource "aws_efs_file_system" "foo" {
  creation_token = "my-product"

  # ...

  backup_policy {
    status = "ENABLED"
  }
}
```

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #14372

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSEFSFileSystemBackupPolicy_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEFSFileSystemBackupPolicy_ -timeout 120m
=== RUN   TestAccAWSEFSFileSystemBackupPolicy_basic
=== PAUSE TestAccAWSEFSFileSystemBackupPolicy_basic
=== RUN   TestAccAWSEFSFileSystemBackupPolicy_disappears
=== PAUSE TestAccAWSEFSFileSystemBackupPolicy_disappears
=== CONT  TestAccAWSEFSFileSystemBackupPolicy_basic
=== CONT  TestAccAWSEFSFileSystemBackupPolicy_disappears
--- PASS: TestAccAWSEFSFileSystemBackupPolicy_basic (65.08s)
--- PASS: TestAccAWSEFSFileSystemBackupPolicy_disappears (78.31s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	80.294s
```

Original PR is https://github.com/hashicorp/terraform-provider-aws/pull/14670

### CODE REVIEW

**Comment by shuheiktgw** [comment]
@DrFaust92 I've made it a separate resource so would you review the PR?

File: website/docs/r/efs_file_system_backup_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+The following arguments are supported:
+
+* `file_system_id` - (Required) The ID of the EFS file system.
+* `backup_policy` - (Required) A file system backup_policy object (documented below).
+
+### Backup Policy Arguments
+For **backup_policy** the following attributes are supported:
+
+* `status` - (Required) A status of the backup policy. Valid values: `Enabled`.
```
Comment (pwlandoll) [inline website/docs/r/efs_file_system_backup_policy.html.markdown:39]: Is there any value here to supporting `Disabled` as well, or would that be the same as an absence of the `efs_file_system_backup_policy` resource?

**Review by pwlandoll — APPROVED** [review]

**Comment by pwlandoll** [comment]
@shuheiktgw - it looks like there are some merge conflicts, resolving those might help to get this reviewed.

**Comment by petewilcock** [comment]
@shuheiktgw It'd be great to see this last conflict hurdle resolved and this merged in 🙏

**Comment by ewbankkit** [comment]
**Note to reviewer:** See [this comment](https://github.com/hashicorp/terraform-provider-aws/pull/19886#issuecomment-864197151) about sweeping automated backups.

**Review by pwlandoll — APPROVED** [review]

**Comment by ewbankkit** [comment]
@shuheiktgw Thanks for the contribution :tada: :clap:.
To match the underlying AWS EFS API I have renamed this new resource to `aws_efs_backup_policy`.

File: website/docs/r/efs_file_system_backup_policy.html.markdown (hunk 1 above)
Comment (ewbankkit) [reply website/docs/r/efs_file_system_backup_policy.html.markdown]: I went ahead and pushed a change to allow `DISABLED`.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSEFSBackupPolicy_'                   
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEFSBackupPolicy_ -timeout 180m
=== RUN   TestAccAWSEFSBackupPolicy_basic
=== PAUSE TestAccAWSEFSBackupPolicy_basic
=== RUN   TestAccAWSEFSBackupPolicy_disappears_fs
=== PAUSE TestAccAWSEFSBackupPolicy_disappears_fs
=== RUN   TestAccAWSEFSBackupPolicy_update
=== PAUSE TestAccAWSEFSBackupPolicy_update
=== CONT  TestAccAWSEFSBackupPolicy_basic
=== CONT  TestAccAWSEFSBackupPolicy_update
=== CONT  TestAccAWSEFSBackupPolicy_disappears_fs
--- PASS: TestAccAWSEFSBackupPolicy_disappears_fs (21.46s)
--- PASS: TestAccAWSEFSBackupPolicy_basic (27.39s)
--- PASS: TestAccAWSEFSBackupPolicy_update (49.78s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	53.322s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSEFSBackupPolicy_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSEFSBackupPolicy_ -timeout 180m
=== RUN   TestAccAWSEFSBackupPolicy_basic
=== PAUSE TestAccAWSEFSBackupPolicy_basic
=== RUN   TestAccAWSEFSBackupPolicy_disappears_fs
=== PAUSE TestAccAWSEFSBackupPolicy_disappears_fs
=== RUN   TestAccAWSEFSBackupPolicy_update
=== PAUSE TestAccAWSEFSBackupPolicy_update
=== CONT  TestAccAWSEFSBackupPolicy_basic
=== CONT  TestAccAWSEFSBackupPolicy_update
=== CONT  TestAccAWSEFSBackupPolicy_disappears_fs
--- PASS: TestAccAWSEFSBackupPolicy_disappears_fs (24.32s)
--- PASS: TestAccAWSEFSBackupPolicy_basic (31.82s)
--- PASS: TestAccAWSEFSBackupPolicy_update (60.52s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	63.590s
```

**Pull request merged**